docs/structured-streaming-programming-guide.md

```diff
@@ -1225,6 +1225,12 @@ Note that there are some restrictions when you use session window in streaming q
 
 For batch query, global window (only having `session_window` in grouping key) is supported.
 
+Due to implementation details of session window in streaming query, for each input row in same session,
```

Comment:
Why not explain the behavior of default and which case it helps? If we don't explain the case default is better, we fail to let end users reason about why this config exists, as well as why we disable this by default.

How about revising the content like this?

````
By default, Spark does not perform partial aggregation for session window aggregation, since it requires additional sort in local partitions before grouping. It works better for the case there are only few number of input rows in same group key for each local partition, but for the case there are numerous input rows having same group  key in local partition, doing partial aggregation can still increase the performance significantly despite additional sort.

You can enable `spark.sql.streaming.sessionWindow.merge.sessions.in.local.partition` to indicate Spark to perform partial aggregation.
```

Reply:
I'm fine with revised content. The only point I have is to document it.

Reply:
Revised based on your comment. If no question, I will merge it.